### ISSUE TITLE
Support for Traefik's Forward Authentication

### ISSUE THREAD

**Issue by gheibia** [lawrencegripper/traefik-appinsights-watchdog#27]
I've recently deployed a service which allows me to take advantage of Traefik's Forward Authentication and secure its API endpoint with Azure AD (https://github.com/thomseddon/traefik-forward-auth/pull/67). The service employes Authorization Code flow. The way it works is that upon successful completion of the auth flow with AD, the service sets an Auth token and redirect the client back to the originally requested endpoint (e.g.: /dashboard).
Since the auth flow requires user interaction, Watchdog won't be able to do health-checks on its own.

What I'm proposing (and will be happy to implement) is to introduce Client Credential flow support in watchdog yielding an access token which can then be exchanged with a cookie from the new Auth service. The cookie can be used to call Traefik's "/health" endpoint.

What will change here will be the addition of 3 new parameters: Client ID, Client Secret and the Auth Service endpoint for exchange. I don't think we need Tenant ID if we use version 1 of Microsoft Identity Platform's APIs.

Obviously, Watchdog will require an App Registration in AD with an Admin already provided the consent (consent acquisition won't be part of this).

Any feedback will be appreciated.

**Comment by lawrencegripper** [comment]
Yeah I'm happy for you to pick this up and make the contribution. Lets ensure, you would have anyway, that these are optional and current behavior is maintained unless this option is enabled. 

As a side note I'm not actively developing or maintaining this code anymore so I'll be updating the readme to note this, that doesn't mean anything will change about it's current operation but one option worth considering is that you could take a fork for this work and therefore not be reliant on my review etc (which could be slow as I have other priorities at the moment).

**Comment by gheibia** [comment]
Thanks for the heads up. I will have to fork this anyway to create a PR, but won't wait for it to get merged.
